src/haddock/modules/refinement/openmm/__init__.py

```diff
@@ -0,0 +1,114 @@
+"""OpenMM refinement module for HADDOCK3."""
+import os
+import subprocess
+from contextlib import suppress
+from pathlib import Path
+
+from haddock.libs.libontology import PDBFile
+# from haddock.libs.libparallel import Scheduler
+from haddock.modules import BaseHaddockModule, get_engine
+
+
+# allow general testing when OpenMM is not installed
+with suppress(ImportError):
+    from haddock.modules.refinement.openmm.openmm import OPENMM
+
+
+RECIPE_PATH = Path(__file__).resolve().parent
+DEFAULT_CONFIG = Path(RECIPE_PATH, "defaults.yaml")
+
+
+def runSubprocess(command_to_run):
+    """Run subprocess."""
+    outputOfSubprocess = subprocess.run([command_to_run],
+                                        shell=True,
+                                        capture_output=True,
+                                        encoding='utf-8')
+    return outputOfSubprocess.stdout.strip()
+
+
+class HaddockModule(BaseHaddockModule):
+    """HADDOCK3 OpenMM module."""
+
+    name = RECIPE_PATH.name
+
+    def __init__(self, order, path, initial_params=DEFAULT_CONFIG):
+        super().__init__(order, path, initial_params)
+
+    def create_directories(self):
+        """Create the necessary directories and provides the paths."""
+        directory_list = [
+            "pdbfixer",
+            "solvation_boxes",
+            "intermediates",
+            "md_raw_output",
+            "openmm_output"
+            ]
+
+        directory_dict = {}
+        for dir in directory_list:
+            self.log(f"Creating directory {dir}")
+            os.mkdir(dir)
+            directory_dict[dir] = dir
+        
+        return directory_dict
+
+    @classmethod
+    def confirm_installation(cls):
+        """Confirm installation of openmm and pdfixer."""
+        checkOpenMM = runSubprocess("conda list openmm --json")
+        checkPdbfixer = runSubprocess("conda list pdbfixer --json")
+
+        if (checkOpenMM == '[]'):
+            raise Exception('OpenMM is not installed in conda.')
+        if (checkPdbfixer == '[]'):
+            raise Exception('OpenMM pdbfixer is not installed in conda.')
+        return
+
+    def _run(self):
+        """Execute module."""
+        previous_models = self.previous_io.retrieve_models(
+            individualize=True
+            )
+
+        previous_models.sort()
+        models_to_export = []
+        # create directories
+        directory_dict = self.create_directories()
+
+        openmm_jobs = []
+        for i, model_to_be_simulated in enumerate(previous_models, start=1):
+            openmm_jobs.append(
+                OPENMM(
+                    identificator=i,
+                    model=model_to_be_simulated,
+                    path=Path("."),
+                    params=self.params,
+                    directory_dict=directory_dict
+                    )
+                )
+
+        # running jobs
+        # currently it only accepts local and mpi.

```

Comment:
Any clue why `hpc` mode is not accepted?

Reply:
Likely this is the last thing missing in this PR :+1: :wink:

Reply:
I checked. The problem can be solved running openmm in local mode on gpu architectures. The openmm infrastructure is able to detect the available platforms and to choose the faster among them! This is cool and makes the module far more usaful!